DeviceAdministration/Infrastructure/Repository/ActionRepository.cs

```diff
@@ -11,12 +11,35 @@ namespace Microsoft.Azure.Devices.Applications.RemoteMonitoring.DeviceAdmin.Infr
     /// </summary>
     public class ActionRepository : IActionRepository
     {
+        private readonly HttpMessageHandler _handler;
+
+        public ActionRepository(HttpMessageHandler handler = null)
+        {
```

Comment:
why `HttpMessageHandler handler = null`? 
and did you test dependency injection? how does it know to use inject HttpMessageHandlerStub as  HttpMessageHandler ?